### ISSUE TITLE
XGBoost 초기 점수 짝 실행을 Kaggle CPU에서 미리 만든다

### ISSUE THREAD

**Issue by tmheo** [tmheo/predicting-smartphone-addiction#524]
## Question

동결된 `exp111_xgb_depth8_no_te` 기준과 `exp211_issue520_xgb_lr_onehot_init` 후보의 정식 실행 산출물을 Kaggle CPU에서 같은 공급자와 실행 환경 등급으로 미리 만들고, 메인 MLflow에 검증 가능한 실행 기록 묶음으로 반입해 후속 최종 판정이 즉시 소비할 수 있게 할 수 있는가?

- `artifacts/issue520-initial-score-extension-precommit.json`의 설정, 분할, 시드와 해시를 바꾸지 않는다.
- 기준 seed 42와 후보 seed 42를 같은 Kaggle CPU 실행 환경 등급에서 실행한다.
- 후보가 seed 42 짝차이 양수와 진입 하한을 통과하면 후보 시드 42, 43, 44 확정 실행도 같은 Kaggle CPU 실행 환경 등급에서 수행한다.
- 표준 `pipeline.run`, `pipeline.bundle export`, 실행 기록 묶음 반입 및 검증 경로만 사용한다.
- 기준과 후보의 입력 해시, 커밋, 깨끗한 상태, 공급자 태그와 실행 환경 등급을 검증한다.
- 모든 실행을 메인 MLflow에 반입하고 설정, OOF, 시드별 OOF, 시험 예측, 제출 파일과 초기 점수 근거 산출물이 열리는지 확인한다.
- 후보 풀 판정, 등록 또는 기각은 수행하지 않는다.

완료 산출물은 메인 MLflow의 기준 및 후보 실행 식별자, 조건부 확정 실행 식별자와 반입 검증 근거다.

**Comment by tmheo** [comment]
## 해결

동결된 XGBoost 기준과 초기 점수 후보의 Kaggle CPU 사전 계산, 조건부 3시드 확인, 메인 MLflow 반입과 산출물 검증을 완료했다.

### 메인 MLflow 실행

| 역할 | 원격 실행 식별자 | 메인 MLflow 실행 식별자 | OOF AUC |
|---|---|---|---:|
| `exp111_xgb_depth8_no_te` seed 42 기준 | `9fb8aeb15be546b6a31a447b7b686493` | `6252f39ab8d34d7eb05764504c54bcc5` | `0.964401626063` |
| `exp211_issue520_xgb_lr_onehot_init` seed 42 후보 | `e7f5d9a5abd946c0b92d17ba268725cf` | `10c4a8e883784dd2b8fff1ee47b1fb7e` | `0.967806381540` |
| 같은 후보 seed 42, 43, 44 확인 | `b78631d2bb8f49d18c534566956b4d4a` | `ee4989467d3f453e8e99d80c89d39ae9` | `0.967941256400` |

### 조건부 실행 판정

- seed 42 후보와 기준의 짝차이는 `+0.003404755477`이고 바깥쪽 분할은 `5/5`에서 후보가 높았다.
- 후보 seed 42 AUC `0.967806381540`은 진입 하한 `0.959367610562`를 통과했다.
- 동결 조건을 통과했으므로 후보 seed 42, 43, 44를 재사용 없이 모두 정식 실행했다.
- 시드별 AUC는 seed 42 `0.967806381540`, seed 43 `0.967808182709`, seed 44 `0.967796395492`다.
- 3시드 평균본의 최근접 구성원은 `exp209_issue505_lgb_lr_onehot_init`, 스피어만 상관은 `0.997474581488`로 중복 문턱 `0.998` 미만이다.
- 이 티켓의 범위에서는 후보 풀 판정, 등록, 기각과 `pool.yaml` 변경을 하지 않았다.
  후속 판정은 [XGBoost 정확값 선형 OOF 로짓 후보를 갱신된 풀에서 최종 판정한다](https://github.com/tmheo/predicting-smartphone-addiction/issues/522)가 수행한다.

### 반입 및 무결성 근거

- 세 실행 모두 출처 커밋 `bf0fd22fd1fe541fd969e7526161da61b68887d0`, 깨끗한 Git 상태, `remote.provider=kaggle`, 실행 환경 등급 `cpu`를 기록했다.
- 세 실행의 Kaggle 이미지가 모두 `gcr.io/kaggle-images/python@sha256:dafd4ce5668bbf1ad422e4c109e0f18c9623c3a7c7f48b0235f13142755c40b9`로 같았다.
- 세 실행의 `uv.lock` SHA-256은 모두 `27527bf7a3094af0c9fa85613216f097a9d18b4cf75ab4c5b4c73b2ad3af25b4`다.
- 실행 환경은 Python `3.13.13`, numpy `2.5.2`, pandas `3.0.5`, scikit-learn `1.9.0`, XGBoost `3.4.0`, MLflow `3.15.1`로 기록했다.
- 입력 SHA-256은 train `f4669147311c76eb03496061a852af283efcf0f12cf5c19274e775def81edd9c`, test `8b462dd47fe8165cd0b082bf33b56523c5811453070af48b9f86b2eb928de49e`, folds `5f5d09e9356f227ecb4a063270b175bb5cae20afb25636c563db185e18a155c4`로 모두 같았다.
- 실행 기록 묶음 SHA-256은 기준 선별 `dcd575c1a7a8816f96368df0c6b93fa44910cf2bac865d241a1d23ce67ec1d35`, 후보 선별 `a8e6b5f933e5e1ed9e2f10ca312b2b8061c4ce35e330196bfdb97b73d9bb78e2`, 후보 확인 `3776dd32644d0b4ff0addc3cb9945f4651394f943610f1fa1b31a67d342641c8`이며 원격 기록값과 내려받은 파일이 일치했다.
- 세 묶음은 입력 해시, 출처 커밋, 깨끗한 상태, 설정 해시와 시드별 OOF 재채점 관문을 통과해 `/Users/taemyungheo/workspace/github/kaggle/predicting-smartphone-addiction/mlflow.db`에 반입됐다.
- 메인 MLflow에서 각 실행의 설정, OOF, 시드별 OOF, 시험 예측, 제출 파일과 실행 환경 기록을 실제로 열었다.
  후보 선별의 초기 점수 근거 5개 좌표와 확인 실행의 15개 시드 및 바깥쪽 분할 좌표도 모두 열어 완전성을 확인했다.
- 후보 seed 42 OOF는 선별 실행과 확인 실행에서 SHA-256 `9e20d8f43d864cb6aca139207b6897d02b323548c11737eb0428b57046528902`로 동일했다.
- 로컬 판정 기록 해시는 선별 `9e00dbaab9b5ae9b4c9241bcfaf207ec965b2617bb8bd7e8871e546da10ba0c8`, 확인 `526936c31921688fb7b714564a45fb39f843701122cd8fe6535b8f07d5734b6d`다.
- Kaggle 공개 점수와 제출 결과는 사용하지 않았다.
